package.json

```diff
@@ -39,6 +39,7 @@
       "@overnightjs/core": "^1.7.5",
       "aws-cron-parser": "1.1.5",
       "base64url": "^3.0.1",
+      "bloomfilter.js": "^1.0.0",
```

Comment:
Went with [this one](https://www.npmjs.com/package/bloomfilter.js) mostly because it's MIT license and had serialization functionality.  Didn't do much however in terms of verifying that the implementation is solid.  Not sure if there's any good way to get signal on that.

Reply:
What were the alternatives you found? Iirc last time I looked there wasn't that many around. 
Maybe we should just use the eth bloom filter? https://github.com/hyperdivision/eth-bloomfilter
That should be well documented and implemented in other langs. Codebase is minimal and maintained by a really solid dev.

Reply:
Looking at the code I see two problems with using that one.  The first more minor issue is that the size of the bloom filter is fixed at 256 bits, it doesn't scale with the size of the input or the desired false positive rate.  The bigger issue is that it doesn't have any serialization/deserialization functionality, which we need to persist the filter into the `TreeMetadata` ipfs record.

Reply:
**Serialization** it uses a buffer which can easily be serialized/deserialized

**Size** I guess we should figure out if 256 *bytes* is actually a problem?

Reply:
But the internal buffer isn't exposed.  Are you suggesting we make a PR to that library to allow constructing the bloom filter from a given buffer?

Reply:
yes it's exposed on a public property `.buffer`.

Reply:
New version up that switches back to the originally proposed library: https://www.npmjs.com/package/bloom-filters.
Suppressing the compilation warning worked trivially